### ISSUE TITLE
v-show doesn't not work for components on server-side rendering

### ISSUE THREAD

**Issue by fenivana** [vuejs/vue#5239]
### Version
2.2.4

### Reproduction link
[https://github.com/fenivana/ssr-v-show-issue](https://github.com/fenivana/ssr-v-show-issue)

### Steps to reproduce
```
npm install
node server.js
```

### What is expected?
```html
<div server-rendered="true"><div style="display:none">hello world</div></div>
```

### What is actually happening?
```html
<div server-rendered="true"><div>hello world</div></div>
```

**Comment by znck** [comment]
`package.json` is missing in repro.

**Comment by Kingwl** [comment]
looks fixed via #5224

**Comment by fenivana** [comment]
@znck Sorry, my fault. Added.

**Comment by znck** [comment]
Apparently fixed in #5224. If it persists, reopen the issue.
